File: android/src/main/java/com/example/mobile_im_plugin/IMClientManager.java
package com.example.mobile_im_plugin;

import android.content.Context;

import com.example.mobile_im_plugin.events.ChatBaseEventImpl;
import com.example.mobile_im_plugin.events.ChatMessageEventImpl;
import com.example.mobile_im_plugin.events.MessageQoSEventImpl;

import net.x52im.mobileimsdk.android.ClientCoreSDK;
import net.x52im.mobileimsdk.android.conf.ConfigEntity;

import io.flutter.plugin.common.MethodChannel;

public class IMClientManager {
    private static String TAG = IMClientManager.class.getSimpleName();

    private static IMClientManager instance = null;

    /**
     * MobileIMSDK是否已被初始化. true表示已初化完成，否则未初始化.
     */
    private boolean init = false;

    /**
     * 基本连接状态事件监听器
     */
    private ChatBaseEventImpl baseEventListener = null;
    /**
     * 数据接收事件监听器
     */
    private ChatMessageEventImpl transDataListener = null;
    /**
     * 消息送达保证事件监听器
     */
    private MessageQoSEventImpl messageQoSListener = null;


    public static IMClientManager getInstance() {
        if (instance == null)
            instance = new IMClientManager();
        return instance;
    }

    private IMClientManager() {
//        initMobileIMSDK();
    }

    /**
     * MobileIMSDK的初始化方法。正式的APP项目中，建议本方法在Application的子类中调用。
     */
    public void initMobileIMSDK(Context context, MethodChannel channel, String appKey, String serverIP, int serverPort, boolean debug) {
        if (!init) {
            // 设置ConfigEntity
            ConfigEntity.appKey = appKey;
            ConfigEntity.serverIP = serverIP;
            ConfigEntity.serverPort = serverPort;

            // MobileIMSDK核心IM框架的敏感度模式设置
//			ConfigEntity.setSenseMode(SenseMode.MODE_15S);

            // 设置最大TCP帧内容长度（不设置则默认最大是 6 * 1024字节）
//			LocalSocketProvider.TCP_FRAME_MAX_BODY_LENGTH = 60 * 1024;

            // 开启/关闭DEBUG信息输出
            ClientCoreSDK.DEBUG = debug;

            // 【特别注意】请确保首先进行核心库的初始化（这是不同于iOS和Java端的地方)
            ClientCoreSDK.getInstance().init(context);

            // 设置事件回调
            baseEventListener = new ChatBaseEventImpl(channel);
            transDataListener = new ChatMessageEventImpl(channel);
            messageQoSListener = new MessageQoSEventImpl(channel);
            ClientCoreSDK.getInstance().setChatBaseEvent(baseEventListener);
            ClientCoreSDK.getInstance().setChatMessageEvent(transDataListener);
            ClientCoreSDK.getInstance().setMessageQoSEvent(messageQoSListener);
            init = true;
        }
    }

    public void release() {
        ClientCoreSDK.getInstance().release();
        resetInitFlag();
    }


    /**
     * 重置init标识。
     * <p>
     * <b>重要说明：</b>不退出APP的情况下，重新登陆时记得调用一下本方法，不然再
     * 次调用 {@link #initMobileIMSDK()} 时也不会重新初始化MobileIMSDK（
     * 详见 {@link #initMobileIMSDK()}代码）而报 code=203错误！
     */
    public void resetInitFlag() {
        init = false;
    }

    public ChatMessageEventImpl getTransDataListener() {
        return transDataListener;
    }

    public ChatBaseEventImpl getBaseEventListener() {
        return baseEventListener;
    }

    public MessageQoSEventImpl getMessageQoSListener() {
        return messageQoSListener;
    }

}

File: ios/Classes/IMClientManager.m
#import "IMClientManager.h"
#import "ClientCoreSDK.h"
#import "ConfigEntity.h"
#import "TCPFrameCodec.h"


///////////////////////////////////////////////////////////////////////////////////////////
#pragma mark - 私有API
///////////////////////////////////////////////////////////////////////////////////////////

@interface IMClientManager ()

/* MobileIMSDK是否已被初始化. true表示已初化完成，否则未初始化. */
@property (nonatomic) BOOL _init;
//
@property (strong, nonatomic) ChatBaseEventImpl *baseEventListener;
//
@property (strong, nonatomic) ChatMessageEventImpl *transDataListener;
//
@property (strong, nonatomic) MessageQoSEventImpl *messageQoSListener;

@property (strong, nonatomic) FlutterMethodChannel *channel;

@end


///////////////////////////////////////////////////////////////////////////////////////////
#pragma mark - 本类的代码实现
///////////////////////////////////////////////////////////////////////////////////////////

@implementation IMClientManager

// 本类的单例对象
static IMClientManager *instance = nil;

+ (IMClientManager *)sharedInstance {
    if (instance == nil) {
        instance = [[super allocWithZone:NULL] init];
    }
    return instance;
}

/*
 *  重写init实例方法实现。
 *
 *  @return
 *  @see [NSObject init:]
 */
- (id)init {
    if (![super init]) return nil;
    return self;
}

- (void)initMobileIMSDKWithChannel: (FlutterMethodChannel *)channel appKey: (NSString *)appKey serverIP: (NSString *)serverIP serverPort: (int)serverPort debug: (BOOL)debug  {
    if(!self._init) {
        // 设置AppKey 服务器IP和服务器端口
        [ConfigEntity registerWithAppKey:appKey];
        [ConfigEntity setServerIp:serverIP];
        [ConfigEntity setServerPort:serverPort];
        
        // 使用以下代码表示不绑定固定port（由系统自动分配），否则使用默认的7801端口
        //      [ConfigEntity setLocalSendAndListeningPort:-1];
        
        // RainbowCore核心IM框架的敏感度模式设置
        //      [ConfigEntity setSenseMode:SenseMode10S];
        
        // 设置最大TCP帧内容长度（不设置则默认最大是 6 * 1024字节）
        //      [TCPFrameCodec setTCP_FRAME_MAX_BODY_LENGTH:60 * 1024];
        
        // 开启DEBUG信息输出
        [ClientCoreSDK setENABLED_DEBUG:debug];
        
        // 设置事件回调
        self.baseEventListener = [[ChatBaseEventImpl alloc] initWithChannel: channel];
        self.transDataListener = [[ChatMessageEventImpl alloc] initWithChannel: channel];
        self.messageQoSListener = [[MessageQoSEventImpl alloc] initWithChannel: channel];
        [ClientCoreSDK sharedInstance].chatBaseEvent = self.baseEventListener;
        [ClientCoreSDK sharedInstance].chatMessageEvent = self.transDataListener;
        [ClientCoreSDK sharedInstance].messageQoSEvent = self.messageQoSListener;
        
        self._init = YES;
    }
}

- (void)releaseMobileIMSDK {
    [[ClientCoreSDK sharedInstance] releaseCore];
    [self resetInitFlag];
}

- (void)resetInitFlag {
    self._init = NO;
}

- (ChatMessageEventImpl *) getTransDataListener {
    return self.transDataListener;
}

- (ChatBaseEventImpl *) getBaseEventListener {
    return self.baseEventListener;
}

- (MessageQoSEventImpl *) getMessageQoSListener {
    return self.messageQoSListener;
}

@end
